runtime/src/main/java/io/micronaut/jackson/bind/MapToObjectConverter.java

```diff
@@ -58,22 +59,27 @@ public MapToObjectConverter(Provider<BeanPropertyBinder> beanPropertyBinder) {
 
     @Override
     public Optional<Object> convert(Map map, Class<Object> targetType, ConversionContext context) {
-        if (targetType.isInstance(map)) {
+        final Function<Map, Function<Object, Object>> propertiesBinderFunction = properties -> object -> {
+            Map<?, ?> theMap = properties;
+            Map bindMap = new LinkedHashMap(theMap.size());
+            for (Map.Entry<?, ?> entry : theMap.entrySet()) {
+                Object key = entry.getKey();
+                bindMap.put(NameUtils.decapitalize(NameUtils.dehyphenate(key.toString())), entry.getValue());
+            }
+            return beanPropertyBinder.get().bind(object, bindMap);
```

Comment:
Why isn't the bean introspection API being used here if its available?

Reply:
Because [this](https://github.com/micronaut-projects/micronaut-core/blob/refactorMapToObjectConverter/validation/src/test/groovy/io/micronaut/validation/BookmarkControllerSpec.groovy#L102) test fails as the `parameters` Map generated by above is:
```
[sort: ['href'], order: ['desc'], max: ['10'], offset: ['0'], ids:[1] ]
```
And I am not sure how to assign them using `BeanProperty` using `Introspection` API which throws conversion error for not being able to convert `[1]` to `Integer` as property `max`, `offset` are `Integer` and not `List`.